pandas/tests/indexes/test_category.py

```diff
@@ -365,18 +365,18 @@ def test_astype(self):
         tm.assert_index_equal(result, expected)
 
     def test_reindex_base(self):
-
-        # determined by cat ordering
-        idx = self.create_index()
+        # Determined by cat ordering.
+        idx = CategoricalIndex(list("cab"), categories=list("cab"))
         expected = np.arange(len(idx), dtype=np.intp)
 
         actual = idx.get_indexer(idx)
         tm.assert_numpy_array_equal(expected, actual)
```

Comment:
@jreback @jorisvandenbossche : This test does exactly what you both were asking for: unique elements tested against identical indexer